src/services/user.service.js

```diff
@@ -21,12 +21,19 @@ import store from '@/store/'
 import Alert from '@/model/Alert.model'
 
 class UserService {
+  /**
+   * @param {string} baseUrl - HTTP base URL used to access JupyterHub REST API
+   */
+  constructor (baseUrl) {
+    this.baseUrl = baseUrl
+  }
+
   /**
    * Gets the user profile from the backend server.
    * @returns {Promise<*>} - a promise that dispatches Vuex action
    */
   getUserProfile () {
-    return axios.get(`${window.location.pathname}/userprofile`).then((response) => {
```

Comment:
I had noticed this was causing a `//` in the URL (which the UIS/hub seemed to handle ok), might be worth using a URL join function in the future.